src/commands/init.ts

```diff
@@ -28,20 +29,85 @@ export class InitCommand implements Command {
 
   run(options, config): Promise<any> {
     // Defer dependency loading until this specific command is run
-    const PolykartGenerator = require('../templates/polykart').PolykartGenerator;
+    const PolykartGenerator =
+        require('../templates/polykart').PolykartGenerator;
+    const findup = require('findup');
+    const inquirer = require('inquirer');
     const YeomanEnvironment = require('yeoman-environment');
 
     return new Promise((resolve, reject) => {
       let logger = logging.getLogger('cli.init');
-      let templateName = options['name'] || 'polymer-init';
 
       logger.debug('creating yeoman environment...');
       let env = new YeomanEnvironment();
-      env.register(require.resolve('generator-polymer-init'), 'polymer-init:app');
-      env.registerStub(PolykartGenerator, 'polykart:app');
+      env.register(
+          require.resolve('generator-polymer-init'), 'polymer-init-basic:app');
+      env.registerStub(PolykartGenerator, 'polymer-init-shop:app');
+
+      env.lookup(() => {
+        let generators = env.getGeneratorsMeta();
+
+        let runGenerator = (generatorName: string, templateName?: string) => {
+          let generator = generators[generatorName];
+          if (generator) {
+            logger.info(`Running template ${templateName || generatorName}`);
+            logger.debug(`Running generator ${generatorName}`);
+            env.run(generatorName, {}, () => resolve());
+          } else {
+            logger.warn('Template ${options.name} not found');
+            logger.debug('Yeoman generator ${generatorName} not found');
+          }
+        };
+
+        let getDisplayName = (generatorName) => {
+          let nameEnd = generatorName.indexOf(':');
+          if (nameEnd == -1) nameEnd = generatorName.length;
+          return generatorName.substring('polymer-init-'.length, nameEnd);
+        }
+
+        if (options.name) {
+          let generatorName = `polymer-init-${options.name}:app`;
+          runGenerator(generatorName, options.name);
+        } else {
+          let polymerInitGenerators = Object.keys(generators)
+              .filter((k) => k.startsWith('polymer-init')
+                  && k !== 'polymer-init:app');
+          let choices = polymerInitGenerators.map((generatorName: string) => {
```

Comment:
Can we combine this with the logic in L72-74 to just iterate over all generators directly (instead of pulling out the names above and then having to get back the generator here).